src/main/kotlin/kweb/html/StorageReceiver.kt

```diff
@@ -13,35 +12,82 @@ import kweb.util.toJson
 class StorageReceiver(val receiver: WebBrowser, val type: StorageType) {
     private val obj = "${type.name}Storage"
 
-    operator fun set(name: String, value: Any) {
-        setString(name, value.toJson())
+    operator fun set(name: String, value: String) {
+        setJson(name, JsonPrimitive(value))
     }
 
-    fun setString(key: String, value: String) {
-        if (value == "") {
+    operator fun set(name: String, value: Int) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Float) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Double) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Short) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Long) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Boolean) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    operator fun set(name: String, value: Char) {
+        setJson(name, JsonPrimitive(value.toString()))
+    }
+
+    operator fun set(name: String, value: Byte) {
+        setJson(name, JsonPrimitive(value))
+    }
+
+    //I don't know if we actually need this one, but it seems like it might be useful at some point.
+    operator fun set(name: String, value: JsonElement) {
+        setJson(name, value)
+    }
+
+    fun setJson(key: String, value: JsonElement) {
```

Comment:
should we just merge this into `fun set(name: String, value: JsonElement)` since they do the same thing?

Reply:
Yes, we should probably merge them.

Also, I'm now realizing we decided to write the class this way, when we were still abstracting away JsonElements and stuff from end users. We could have `fun set(key: String, value: JsonElement)`, and just have end users convert their data to JsonPrimitive(or object or element), instead of us doing it for them. That'd remove a lot of duplication in this class, and in CookieReceiver

Reply:
I think those convenience functions for JsonPrimitives are useful, they do lead to some duplication but I think its ok in this case.